operator/controllers/batch/batchjob_helpers.go

```diff
@@ -0,0 +1,389 @@
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/aws/aws-sdk-go/service/sqs"
+	batch "github.com/cortexlabs/cortex/operator/apis/batch/v1alpha1"
+	"github.com/cortexlabs/cortex/operator/controllers"
+	"github.com/cortexlabs/cortex/pkg/lib/errors"
+	libjson "github.com/cortexlabs/cortex/pkg/lib/json"
+	"github.com/cortexlabs/cortex/pkg/lib/k8s"
+	s "github.com/cortexlabs/cortex/pkg/lib/strings"
+	"github.com/cortexlabs/cortex/pkg/types/clusterconfig"
+	"github.com/cortexlabs/cortex/pkg/types/spec"
+	"github.com/cortexlabs/cortex/pkg/types/userconfig"
+	kbatch "k8s.io/api/batch/v1"
+	kcore "k8s.io/api/core/v1"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	_enqueuerContainerName = "enqueuer"
+)
+
+type batchJobStatusInfo struct {
+	QueueExists     bool
+	EnqueuingStatus EnqueuingStatus
+	WorkerJob       *kbatch.Job
+}
+
+// EnqueuingStatus is an enum for the different possible enqueuing status
+type EnqueuingStatus string
+
+// Possible EnqueuingStatus states
+const (
+	EnqueuingNotStarted EnqueuingStatus = "not_started"
+	EnqueuingInProgress EnqueuingStatus = "in_progress"
+	EnqueuingDone       EnqueuingStatus = "done"
+	EnqueuingFailed     EnqueuingStatus = "failed"
+)
+
+func (r *BatchJobReconciler) checkIfQueueExists(batchJob batch.BatchJob) (bool, error) {
+	queueName := r.getQueueName(batchJob)
+	input := &sqs.GetQueueUrlInput{
+		QueueName: aws.String(queueName),
+	}
+	_, err := r.AWS.SQS().GetQueueUrl(input)
+	if err != nil {
+		if aerr, ok := err.(awserr.Error); ok {
+			if aerr.Code() == sqs.ErrCodeQueueDoesNotExist {
+				return false, nil
+			}
+			return false, err
+		}
+	}
+	return true, nil
+}
+
+func (r *BatchJobReconciler) createQueue(batchJob batch.BatchJob) (string, error) {
+	queueName := r.getQueueName(batchJob)
+
+	tags := map[string]string{
+		clusterconfig.ClusterNameTag: r.ClusterConfig.ClusterName,
+		"apiName":                    batchJob.Spec.APIName,
+		"apiID":                      batchJob.Spec.APIId,
+		"jobID":                      batchJob.Name,
+	}
+
+	attributes := map[string]string{
+		sqs.QueueAttributeNameFifoQueue:         "true",
+		sqs.QueueAttributeNameVisibilityTimeout: "60",
+	}
+
+	if batchJob.Spec.DeadLetterQueue != nil {
+		redrivePolicy := map[string]string{
+			"deadLetterTargetArn": batchJob.Spec.DeadLetterQueue.ARN,
+			"maxReceiveCount":     s.Int32(batchJob.Spec.DeadLetterQueue.MaxReceiveCount),
+		}
+
+		redrivePolicyJSONBytes, err := libjson.Marshal(redrivePolicy)
+		if err != nil {
+			return "", err
+		}
+
+		attributes[sqs.QueueAttributeNameRedrivePolicy] = string(redrivePolicyJSONBytes)
+	}
+
+	input := &sqs.CreateQueueInput{
+		Attributes: aws.StringMap(attributes),
+		QueueName:  aws.String(queueName),
+		Tags:       aws.StringMap(tags),
+	}
+	output, err := r.AWS.SQS().CreateQueue(input)
+	if err != nil {
+		return "", err
```

Comment:
When receiving errors from non-cortex packages, it is good to wrap it (to attach the stack trace). There may be other cases of this, this is just the first one I noticed.

Reply:
this is actually not an issue because the controller errors are just logged and are never sent to the user.

A stacktrace is printed when logging the error with  the`logr` package